lightning/backend/local.go

```diff
@@ -1609,186 +1744,307 @@ func (s *sizeProperties) iter(f func(p *rangeProperty) bool) {
 	})
 }
 
+func (local *local) EngineFileSizes() (res []EngineFileSize) {
+	local.engines.Range(func(k, v interface{}) bool {
+		engine := v.(*LocalFile)
+		res = append(res, engine.getEngineFileSize())
+		return true
+	})
+	return
+}
+
 type LocalWriter struct {
```

Comment:
PTAL @Little-Wallace, the LocalWriter is basically entirely rewritten, there may be some performance implication of the original code that I may have missed. Thanks.